src/object/player.cpp

```diff
@@ -514,6 +533,72 @@ Player::update(float dt_sec)
         if (m_physic.get_velocity_y() > -350.f && m_controller->hold(Control::UP))
           m_physic.set_velocity_y(-350.f);
       }
+
+      if (m_bubble_timer.check())
+      {
+        Vector beak_local_offset(30.f, 0.0f);
+        float big_offset_x = is_big() ? 4.0f : 0.0f;
+
+        // Calculate the offsets based on the sprite angle
+        float offset_x = std::cos(m_swimming_angle) * 10.0f;
+        float offset_y = std::sin(m_swimming_angle) * 10.0f;
+
+        // Rotate the beak offset based on the sprite's angle
+        float rotated_beak_offset_x = beak_local_offset.x * std::cos(m_swimming_angle) - beak_local_offset.y * std::sin(m_swimming_angle);
+        float rotated_beak_offset_y = beak_local_offset.x * std::sin(m_swimming_angle) + beak_local_offset.y * std::cos(m_swimming_angle);
+
+        Vector player_center = m_col.m_bbox.get_middle();
+        Vector beak_position;
+
+        // Determine direction based on the radians
+        if (std::abs(m_swimming_angle) > static_cast<float>(math::PI_2)) // Facing left
+        {
+          beak_position = player_center + Vector(rotated_beak_offset_x - big_offset_x * 2, rotated_beak_offset_y);
+        }
+        else // Facing right (including straight up or down)
+        {
+          beak_position = player_center + Vector(rotated_beak_offset_x - 4.0f + big_offset_x, rotated_beak_offset_y);
+        }
+
+        int num_bubbles = graphicsRandom.rand(1, 3);
+
+        std::optional<std::vector<SurfacePtr>> bubble_surfaces = m_bubbles_sprite->get_action_surfaces("normal");
+
+        if (bubble_surfaces)
+        {
+          const std::vector<SurfacePtr>& surfaces = bubble_surfaces.value();
+          int surfaces_size = static_cast<int>(surfaces.size());
+
+          for (int i = 0; i < num_bubbles; ++i)
+          {
+            int random_index = graphicsRandom.rand(1, surfaces_size - 1);
+            SurfacePtr bubble_surface = surfaces.at(random_index);
+
+            Vector bubble_pos(0.f, 0.f);
+            if (std::abs(m_swimming_angle) > static_cast<float>(math::PI_2)) // Facing left
+            {
+              bubble_pos = beak_position + Vector(offset_x - big_offset_x * 2, offset_y);
+            }
+            else // Facing right (including straight up or down)
+            {
+              bubble_pos = beak_position + Vector(offset_x - 4.0f + big_offset_x, offset_y);
+            }
+
+            if (num_bubbles > 1)
+            {
+              float burst_offset_x = graphicsRandom.randf(-5.0f, 5.0f);
+              float burst_offset_y = graphicsRandom.randf(-5.0f, 5.0f);
+              bubble_pos.x += burst_offset_x;
+              bubble_pos.y += burst_offset_y;
+            }
+
+            m_active_bubbles.push_back({ bubble_surface, bubble_pos });
```

Comment:
The bubble should not try to spawn if it's not submerged, or else it will spawn the bubble for one frame and then pop in the next frame. To reproduce this bug, get in this position.
![image](https://github.com/user-attachments/assets/48303d54-e006-4751-9594-41e66f3874ed)